src/main/java/ch/njol/skript/effects/EffParse.java

```diff
@@ -0,0 +1,117 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.classes.Parser;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.ExprParseError;
+import ch.njol.skript.lang.*;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.log.ParseLogHandler;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Name("Parse")
+@Description("Parses a string or a list of strings as a type. If \"try to\" is used, " +
+	"the existing value won't be deleted if the attempt to parse fails.")
+@Example("""
+	set {_a::*} to "1", "2a", "3", "4c", "5"
+	parse {_a::*} as integer
+	send {_a::*} # would send 1, 3 and 5
+	send last parse errors # would print errors about 2a and 4c
+	""")
+@Since("INSERT VERSION")
+public class EffParse extends Effect {
```

Comment:
This needs support for type hints. If `try to` is used, you can use add to the existing hints rather than overwrite them. Could even use hints to error at parse time when using this syntax on variables that can't be strings.